src/math_funcs.jl

```diff
@@ -0,0 +1,77 @@
+# define maths function for rotations types
+import Base.(*), Base.(.*), Base.(-)
+
+function add_maths(rot_type)
+
+    # create an expression for a tuple X[1], X[2], etc...
+    fields = fieldnames(rot_type)
+
+    # the code block we are creating
+    qb = quote; end
+
+    # allow rotating a Vec{3, ...} using *    
+    if !(rot_type <: FixedArray)
+        qn = quote
+            (*){T <: Real}(R::$(rot_type), X::Vec{3,T}) = rotate_point(R, X)
```

Comment:
So we can multiply `Quaternion`s by `Vec{3}`s? And so on?

Somehow I'm getting confused... we imported `Quaternion` as a type of number... we shouldn't mess with the semantics of `*` even if it is cool for our application (e.g. someone else might want to use `Quaternion` in an algebraic sense, make a `Vec{3}` of them and multiply that by a scalar quaternion - while they use *Rotations.jl* for some other purpose). This *might* make sense for custom-defined unit quaternions (as opposed to generic quaternions) and for `AngleAxis`, etc, but I'd need convincing.

This isn't exactly @awbsmith's fault!! JuliaGeometry should have `UnitQuaternion` only while *Quaternions.jl* should be in a more generic place for doing quaternion algebra (as opposed to quaternion *rotations*).